### ISSUE TITLE
Automatic Token Issuance

### ISSUE THREAD

**Issue by cryptonaut420** [tokenly/swapbot#15]
This is a low priority item, but another feature from the prototype which would be cool to have (and I'm actively making use of it already). Also something which we might want to have disabled for the public and enabled only for people running their own instance, because it requires the vending address to have actual ownership of the token being issued.

User would be able to add one or more "auto issuances" to their vending addresses, which are triggered once the available supply of the token is empty. Choose a token name "inflation type" and "inflation modifier".

from prototype documentation:

>There are three inflation modes available (which also change what "inflation_modifier" does):
>
>* "as_needed" - this mode creates only as many tokens as needed for whatever pending outgoing transactions, multiplied by the inflation_modifier. e.g 1,000 new tokens are needed, so only 1,000 new tokens are created.
>* "fixed" - this creates a fixed amount of new tokens. If inflation_modifier was set to 50,000, it would simply create 50,000 new tokens
>* "percent" - this increases the total existing token supply by X percent (inflation_modifier). e.g if inflation_modifier is set to 0.10 (10%) and there is already 100,000 tokens out in the wild, then an additional 10,000 would be issued.

**Comment by AdamBLevine** [comment]
One thing I've been thinking about is how useful it would be to tie automatic price increases to issuance events, so that instead of a low price meaning the vending machine if unnnoticed can produce a crazy amount, instead it would automatically hike the price by 10% each time it has to make more to try and maximize its swaps since demand is obviously high.

I think this, if communicated properly, would be a really useful and valuable feature for new coins and tokens

**Comment by cryptonaut420** [comment]
Definitely, that is something that would tie in with the "custom function" price type #14 

We could maybe even have some predefined custom pricing functions that people could choose from and just plug in a few values e.g increase price by X% on issuance

**Comment by AdamBLevine** [comment]
Right, i'm definitely seeing that as we come up with interesting types of configurations making those available as templates.

I have no experience programming besides html and qbasic in middle school, how difficult would it be to teach me to make custom bots?  If it's approachable I can definitely see holding some "most innovative or profitable bot logic" contests to flesh out the early ecosystem

**Comment by cryptonaut420** [comment]
Depends on how complicated you want to get I guess :p should be doable though
